lib/vast/Tower/Link.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2024-present, Trail of Bits, Inc.
+
+#include "vast/Tower/Link.hpp"
+
+#include <ranges>
+
+namespace vast::tw {
+
+    op_mapping reverse_mapping(const op_mapping &from) {
+        op_mapping out;
+        for (const auto &[root, ops] : from) {
+            for (auto op : ops) {
+                out[op].push_back(root);
+            }
+        }
+        return out;
+    }
+
+    void dbg(const op_mapping &mapping, auto &outs) {
+        outs << "Mapping:\n";
+
+        auto flag = mlir::OpPrintingFlags().skipRegions();
+        auto render_op = [&](operation op) -> decltype(outs) & {
+            outs << op << ": ";
+            op->print(outs, flag);
+            if (!op->getRegions().empty())
+                outs << " ... regions ...";
+            return outs;
+        };
+
+        for (const auto &[from, to] : mapping) {
+            outs << "== ";
+            render_op(from) << "\n";
+            for (auto t : to) {
+                outs << "   -> ";
+                render_op(t) << "\n";
+            }
+        }
+    }
+
+    using loc_to_op_t = llvm::DenseMap< loc_t, operations >;
+
+    void dbg(const loc_to_op_t &mapping, auto &outs) {
+        outs << "loc_to_op\n";
+        for (const auto &[loc, to] : mapping) {
+            outs << ".." << loc << "\n";
+            for (auto t : to) {
+                outs << "   -> " << *t << "\n";
+            }
+        }
+    }
+
+    loc_to_op_t gather_map(location_info_t &li, operation op) {
```

Comment:
this sounds as it performs `gather` and `map` as operation, rename to:
```suggestion
    loc_to_op_t gather_loc_to_op(location_info_t &li, operation op) {
```